apps/emqx/src/emqx_persistent_message.erl

```diff
@@ -22,22 +29,39 @@
 
 -include("emqx_persistent_message.hrl").
 
+-define(optvar_ready, emqx_persistent_message_ready).
+
 %%--------------------------------------------------------------------
 
-init() ->
-    %% Note: currently persistence can't be enabled or disabled in the
-    %% runtime. If persistence is enabled for any of the zones, we
-    %% consider durability feature to be on:
-    Zones = maps:keys(emqx_config:get([zones])),
-    IsEnabled = lists:any(fun is_persistence_enabled/1, Zones),
-    persistent_term:put(?PERSISTENCE_ENABLED, IsEnabled),
-    ?WITH_DURABILITY_ENABLED(begin
-        ?SLOG(notice, #{msg => "Session durability is enabled"}),
-        ok = emqx_ds:open_db(?PERSISTENT_MESSAGE_DB, get_db_config()),
-        ok = emqx_persistent_session_ds_router:init_tables(),
-        ok = initialize_session_ds_state(),
-        ok
-    end).
+-doc """
```

Comment:
This is a temporary solution. I don't want to change EMQX boot sequence in this PR too.